File: 1897-redistribute-characters-to-make-all-strings-equal/1897-redistribute-characters-to-make-all-strings-equal.py
class Solution(object):
    def makeEqual(self, words):
        n = len(words)
        freq = [0] * 26

        for word in words:
            for c in word:
                freq[ord(c) - ord('a')] += 1
            
        for i in range(26):
            if freq[i] % n != 0:
                return False

        return True
        

File: 1897-redistribute-characters-to-make-all-strings-equal/1897-redistribute-characters-to-make-all-strings-equal.cpp
class Solution {
public:
    bool makeEqual(vector<string>& words) {
        int n = words.size();
        vector<int> freq(26, 0);

        for(string word : words) {
            for(char c : word) {
                freq[c - 'a']++;
            }
        }

        for(int i = 0; i<26; i++) {
            if(freq[i]%n) return false;
        }

        return true;
    }
};